trackers/dataset/base.py

```diff
@@ -0,0 +1,87 @@
+import abc
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import supervision as sv
+
+
+# --- Base Dataset ---
+class Dataset(abc.ABC):
```

Comment:
Since we will also be working on a dataset class for our Re-ID training pipeline, let's call this `EvaluationDataset` to avoid confusion.